mmv1/products/compute/api.yaml

```diff
@@ -13480,12 +13480,19 @@ objects:
         name: minPortsPerVm
         description: |
           Minimum number of ports allocated to a VM from this NAT.
+      - !ruby/object:Api::Type::Integer
+        name: maxPortsPerVm
+        description: |
+          Maximum number of ports allocated to a VM from this NAT.
+          This field can only be set when enableDynamicPortAllocation is enabled.
       - !ruby/object:Api::Type::Boolean
         name: enableDynamicPortAllocation
         description: |
           Enable Dynamic Port Allocation.
-          If minPorts is set, minPortsPerVm must be set to a power of two greater than or equal to 32. 
+          If minPortsPerVm is set, minPortsPerVm must be set to a power of two greater than or equal to 32.
           If minPortsPerVm is not set, a minimum of 32 ports will be allocated to a VM from this NAT config.
+          If maxPortsPerVm is set, maxPortsPerVm must be set to a power of two greater than minPortsPerVm.
+          If maxPortsPerVm is not set, a maximum of 65536 ports will be allocated to a VM from this NAT config.
```

Comment:
If this field isn't set, does the API return a default of 65536, or does it see the field as NULL (in some way) but this is the server-side behavior if the field is NULL?

If the API returns a default, then this field (and min_ports_per_vm) should be marked with `default_from_api` in a terraform.yaml file. Example: https://github.com/GoogleCloudPlatform/magic-modules/blob/c4f5dc18583adee759132b88c98c20d259029771/mmv1/products/networkservices/terraform.yaml#L51

Reply:
The second alternative: when the field is not set, a server-side of 65536 is used, but the resource still contains a `null`.